### PR TITLE
Completed tasks 1, 2, 3, 4

### PR DESCRIPTION
**Pull request by zion86**

### CODE REVIEW

File: Exercises/4-count-types.js [hunk 1]
```diff
@@ -1,5 +1,13 @@
 'use strict';
 
-const countTypesInArray = null;
+const countTypesInArray = arr => {
+  const hesh = { };
```
Comment (tshemsedinov) [inline Exercises/4-count-types.js]:
```suggestion
  const hash = {};
```

File: Exercises/4-count-types.js [hunk 2]
```diff
@@ -1,5 +1,13 @@
...
+
+  for (const name of arr) {
```
Comment (tshemsedinov) [inline Exercises/4-count-types.js]:
```suggestion
  for (const value of arr) {
```

File: Exercises/4-count-types.js [hunk 3]
```diff
@@ -1,5 +1,13 @@
...
+    const t = typeof name;
+    (t === 'boolean' || 'string' || 'number') && (hesh[t] = (hesh[t] || 0) + 1);
```
Comment (tshemsedinov) [inline Exercises/4-count-types.js]: Terrible line, try to optimize it, moreover, reread task.

**Comment by zion86** [comment]
Update func

File: Exercises/4-count-types.js [hunk 4]
```diff
@@ -1,5 +1,17 @@
 'use strict';
 
-const countTypesInArray = null;
+const countTypesInArray = arr => {
+  const hesh = {};
+
+  for (const value of arr) {
+    const t = typeof value;
+    if (hesh[t] === undefined) {
```
Comment (tshemsedinov) [inline Exercises/4-count-types.js]:
```suggestion
    if (!hesh[t]) {
```

File: Exercises/4-count-types.js [hunk 5]
```diff
@@ -1,5 +1,17 @@
...
+      hesh[t] = 1;
+    } else {
+      ++hesh[t];
+    }
+  }
```
Comment (tshemsedinov) [inline Exercises/4-count-types.js]: See alternatives https://github.com/HowProgrammingWorks/DataTypes/pull/21#discussion_r324387734

**Pull request closed without merging**